module/VuFind/src/VuFind/Connection/ExternalVuFind.php

```diff
@@ -55,6 +57,15 @@ class ExternalVuFind implements
      */
     protected $baseUrl = null;
 
+    /**
+     * Constructor
+     */
+    public function __construct()
```

Comment:
If the class is going to have a hard dependency on the caching downloader, maybe you should just make that a mandatory constructor parameter instead of adding the exception call to the search method.

Reply:
I see your point that hard dependencies should just be passed in the constructor (assuming they're available at construction time).  But that would go against the pattern used by every existing implementation of CachingDownloaderAwareTrait (where I copied this exception handling) and seemingly against the design pattern it intended.  Seems weird to deviate from it in this one circumstance, although if you feel strongly I will do so.

Reply:
I think the trait is designed not to make the cache mandatory since there might be a situation where a class could fall back to alternate behavior if the cache is not injected. I suspect that the existing implementations didn't change constructors for backward-compatibility reasons. Maybe that's an argument not to change it here, but I don't think the existing precedent is very good, so I wouldn't feel compelled to follow it. :-)

Reply:
Fair enough, I will explore that.

Reply:
Ok, this is done.  I considered keeping the null check (now in the constructor), but apparently passing null is *already* an exception since the type is defined as CachingDownloader (and thus I guess not |void).  Yay for type safety?!